### PR TITLE
feat: collect ckb unlock signatures from verifiers

### PR DESCRIPTION
**Pull request by pygman**

### CODE REVIEW

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts [hunk 1]
```diff
@@ -116,6 +120,23 @@ async function verifyDuplicateMintTx(
   return SigErrorOk;
 }
 
+async function verifyDuplicateUnlockTx(
+  pubKey: string,
+  unlockRecords: unlockRecord[],
+  _txSkeleton: TransactionSkeletonObject,
+): Promise<SigError> {
+  const refTxHashes = unlockRecords.map((mintRecord) => {
+    return mintRecord.id;
```
Comment (mkxbl) [inline offchain-modules/packages/app-multisign-server/src/ckbSigner.ts:129]:
```suggestion
  const refTxHashes = unlockRecords.map((unlockRecord) => {
    return unlockRecord.id;
```

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts [hunk 2]
```diff
@@ -185,13 +206,73 @@ async function verifyMintTx(
...
+  }
+
+  const typeArgs = unlockCells.map((output) => {
+    const typescript = output.cell_output.type;
+    return typescript ? typescript.args : '';
+  });
+
+  if (typeArgs.length > 1) {
+    return new SigError(SigErrorCode.InvalidParams, `unlock recode outputs contains not single typescript`);
+  }
```
Comment (mkxbl) [inline offchain-modules/packages/app-multisign-server/src/ckbSigner.ts:256]: It seems typeArgs.length === unlockCells.length ?

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts [hunk 3]
```diff
@@ -185,13 +206,73 @@ async function verifyMintTx(
...
+
+  for (let i = 0; i < unlockRecords.length; i++) {
+    const unlockRecord = unlockRecords[i];
+    if (unlockRecord.xchain !== ChainType.ETH) {
+      //those chains doesn't verify now
+      continue;
```
Comment (mkxbl) [inline offchain-modules/packages/app-multisign-server/src/ckbSigner.ts:262]: better throw error

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts [hunk 4]
```diff
@@ -291,6 +429,58 @@ async function verifyEthMintTx(mintRecord: mintRecord, output: Cell): Promise<Si
...
+        SigErrorCode.InvalidRecord,
+        `typescript code_hash:${typeScript.code_hash} doesn't match with:${sudtTypescript.codeHash}`,
+      );
+    }
+    if (typeScript.hash_type !== sudtTypescript.hashType) {
+      return new SigError(
+        SigErrorCode.InvalidRecord,
+        `typescript hash_type:${typeScript.hash_type} doesn't match with:${sudtTypescript.hashType}`,
+      );
+    }
```
Comment (mkxbl) [inline offchain-modules/packages/app-multisign-server/src/ckbSigner.ts:468]: just verify Hash(typeScript) === assetId

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts [hunk 5]
```diff
@@ -235,6 +316,63 @@ async function verifyEthMintRecords(records: mintRecord[]): Promise<SigError> {
...
 }
 
+async function verifyEthForUnlockRecords(records: unlockRecord[]): Promise<SigError> {
+  const unlockIds = records.map((record) => {
+    return record.id;
+  });
+  const ckbUnlocks = await SigServer.ckbDb.getCkbUnlockByIds(unlockIds);
+  const ckbUnlockMap: { [k: string]: CkbUnlock } = Object.fromEntries(
+    ckbUnlocks.map((ckbUnlock) => [ckbUnlock.id, ckbUnlock]),
+  );
```
Comment (mkxbl) [inline offchain-modules/packages/app-multisign-server/src/ckbSigner.ts:326]: Should remove this, can not getCkbUnlockByIds when verify unlock(not send unlock tx yet). Verify unlockRecords match burnRecordsFromDb

Comment (pygman) [reply offchain-modules/packages/app-multisign-server/src/ckbSigner.ts]: fixed, removed get unlock from db, get burnTxHash via splite '-' from id

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts (hunk 3 above)
Comment (pygman) [reply offchain-modules/packages/app-multisign-server/src/ckbSigner.ts]: fixed

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts (hunk 2 above)
Comment (pygman) [reply offchain-modules/packages/app-multisign-server/src/ckbSigner.ts]: fixed to verify distinct(hash(typescript)).length > 1

File: offchain-modules/packages/app-multisign-server/src/ckbSigner.ts (hunk 1 above)
Comment (pygman) [reply offchain-modules/packages/app-multisign-server/src/ckbSigner.ts]: fixed

**Pull request merged**